File: problems/Swift/70.爬楼梯.swift
/**
 * 70.爬楼梯 [Easy]
 * Time Complexity: O(n), Space Complexity: O(1)
 * https://leetcode-cn.com/problems/climbing-stairs/
 */

class Solution {
  func climbStairs(_ n: Int) -> Int {
    var p = 0, q = 0, r = 1
    for _ in 1 ... n {
      p = q
      q = r
      r = p + q
    }

    return r
  }
}

File: problems/rust-solutions/src/70.爬楼梯.rs
// 70.爬楼梯 [Easy]
// Time Complexity: O(n), Space Complexity: O(1)
// https://leetcode-cn.com/problems/climbing-stairs/

impl Solution {
  pub fn climb_stairs(n: i32) -> i32 {
    let mut p = 0;
    let mut q = 0;
    let mut r = 1;
    for _ in 1..n + 1 {
      p = q;
      q = r;
      r = p + q;
    }

    return r;
  }
}
